CRM/Contact/Form/Relationship.php

```diff
@@ -453,49 +384,18 @@ public function postProcess() {
       CRM_Case_BAO_Case::createCaseRoleActivity($this->_caseId, $relationshipIds, $params['contact_check'], $this->_contactId);
     }
 
-    // Save notes
     // @todo this belongs in the BAO.
-    if ($this->_action & CRM_Core_Action::UPDATE || $params['note']) {
-      foreach ($relationshipIds as $id) {
-        $noteParams = array(
-          'entity_id' => $id,
-          'entity_table' => 'civicrm_relationship',
-        );
-        $existing = civicrm_api3('note', 'get', $noteParams);
-        if (!empty($existing['id'])) {
-          $noteParams['id'] = $existing['id'];
-        }
-        $noteParams['note'] = $params['note'];
-        $noteParams['contact_id'] = $this->_contactId;
-        if (!empty($existing['id']) || $params['note']) {
-          $action = $params['note'] ? 'create' : 'delete';
-          civicrm_api3('note', $action, $noteParams);
-        }
-      }
+    if (isset($params['note'])) {
+      $this->saveRelationshipNotes($relationshipIds, $params['note']);
```

Comment:
I took another look into what happens and it appear to me that the logic previous logic is not that elegant .. for example the $params['note'] was used but there was no kind of checking to confirm that it is exist or not which . so I refactored this method again with proper checking.